constraint/pkg/client/client_opts.go

```diff
@@ -54,7 +54,11 @@ func validateTargetNames(ts []handler.TargetHandler) []string {
 // Driver defines the Rego execution environment.
 func Driver(d drivers.Driver) Opt {
 	return func(client *Client) error {
-		client.driver = d
+		if d.Name() == "" {
+			return ErrNoDriverName
+		}
+		client.drivers[d.Name()] = d
+		client.driverPriority[d.Name()] = len(client.drivers)
```

Comment:
Not really.

There is the argument that it's harder to keep the state of two different maps in sync, except this function is the only place where these maps get altered.

This is counterbalanced against the fact that it'd be slightly harder to write something like driverForTemplate(), since you wouldn't be able to rely on conventions like a missing priority entry == 0, plus you'd need an additional struct.

In the end it's a wash.

Reply:
If d.Name() already exists as part of client.drivers, could this alter the driverPriority since the total len could be different?

Reply:
Good call, I should have checked for uniqueness.